### PR TITLE
Add lesson 4: Recap loops

### PR DESCRIPTION
**Pull request by fensterzumberg**

### CODE REVIEW

File: lesson4/index.md [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+Enter 1 for deposit, 2 for withdrawal
+1
+Current money on account: 30
+Enter amount:
+30
+Enter 1 for deposit, 2 for withdrawal
+2
+You have no money left, program stopped ...
+```
+### Exercise 4: Create a pattern
```
Comment (alinadenisenko) [inline lesson4/index.md:220]: This one was their homework from lesson 2 :)

**Review by alinadenisenko — COMMENTED** [review]
Cool content 👍

File: lesson4/index.md (hunk 1 above)
Comment (fensterzumberg) [reply lesson4/index.md:220]: Hahahaha. Ok, gonna change it

File: lesson4/index.md [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+Their structure is as follows:
+```java
+for (<INITIALIZATION>; <CONDITION>; <INCREMENT>) {
+    // <CODE TO EXECUTE>
+}
+```
+
+For loops are usually used to run `<CODE TO EXECUTE>` a given amount of times.
+They are using an `INITIALIZATION` statement which is run exactly once at the beginning, 
+a `CONDITION` statement which is run after each iteration and
```
Comment (olayinkasf) [inline lesson4/index.md:32]: condition is run before the iteration

Comment (fensterzumberg) [reply lesson4/index.md:32]: Correct. Fixing

Comment (olayinkasf) [reply lesson4/index.md]: Would be good to point out that the for loop stops once the condition evaluates to false.

**Pull request merged**